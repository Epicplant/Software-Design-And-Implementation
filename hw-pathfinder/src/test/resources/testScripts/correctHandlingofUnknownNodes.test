# Tests that if a node doesn't exist within the graph when Dijkstra's algorithm is applied that the node is shown
# as unknown

# Creates a new graph
CreateGraph newGraph

# Attempts to find a path between two non-existent nodes
FindPath newGraph A B

# Adds node A to the graph
AddNode newGraph A

# Attempts to find a path between one non-existent source node and a dest node
FindPath newGraph B A

# Attempts to find a path between a source node and a non-existent dest node
FindPath newGraph A B