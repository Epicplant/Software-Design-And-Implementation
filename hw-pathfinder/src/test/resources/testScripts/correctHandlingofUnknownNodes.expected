# Tests that if a node doesn't exist within the graph when Dijkstra's algorithm is applied that the node is shown
# as unknown

# Creates a new graph
created graph newGraph

# Attempts to find a path between two non-existent nodes
unknown: A
unknown: B

# Adds node A to the graph
added node A to newGraph

# Attempts to find a path between one non-existent source node and a dest node
unknown: B

# Attempts to find a path between a source node and a non-existent dest node
unknown: B